### PR TITLE
COMPAT: revert to old pandas in imputation.ros

### PR DESCRIPTION
**Pull request by phobson**
Fixes issues compatiibility issues with older versions of pandas.

Namely, removes calls to `DataFrame.pipe` and `.assign`

### CODE REVIEW

**Comment by josef-pkt** [comment]
You will need to rebase and undo the disabling of the tests
https://github.com/statsmodels/statsmodels/pull/3033/files

(I will look again tomorrow, it's late here.)

**Comment by josef-pkt** [comment]
This still fails on 4 out of 6 travis runs.

**Comment by phobson** [comment]
I just pushed some more precise expected values that should take care of most of the failures.

The failures on the most primitive test run (python 2.6?) look to be very fundamental. Should we limit the `ros` module to Python >= 2.7 only?

**Comment by josef-pkt** [comment]
I'll look at the test run a bit later today (in an hour or three)
If it's not trivial, then we skip python 2.6 testing for this. We will drop 2.6 support right after the release.

File: statsmodels/imputation/tests/test_ros.py [hunk 1]
```diff
@@ -1,7 +1,5 @@
 from __future__ import division
 
-__test__ = False
```
Comment (josef-pkt) [inline statsmodels/imputation/tests/test_ros.py]: keep this in here conditional on pandas < 0.14, so we just skip unit tests for older pandas

pandas had limited assert support in <0.14. IIRC, I had problems with older pandas versions at some point and just avoided almost all the pandas testing asserts

**Comment by josef-pkt** [comment]
To get this green, I suggest to skip tests for pandas < 0.14
pandas 0.13 has only problems in test assert

problems with pandas 0.12, on python 2.6, might be more serious, but I think we just ignore them as "ancient"

**Comment by josef-pkt** [comment]
@phobson  I did some final changes and merged in #3054

Thank you

**Pull request closed without merging**